packages/jsx/src/children.ts

```diff
@@ -8,3 +10,15 @@ import type { TSESTree } from "@typescript-eslint/types";
 export function hasChildren(node: TSESTree.JSXElement | TSESTree.JSXFragment) {
   return node.children.length > 0;
 }
+
+export function isChildOfHtmlElement(node: TSESTree.Node) {
```

Comment:
Considering that `@eslint-react/jsx` is intended to be a framework-agnostic library.

There is no such thing as `HTML` in `JSX` nor in `React`.

When an element `type` starts with a lowercase letter, it refers to a `built-in component` in JSX (which is then represented as [HostComponent](https://github.com/facebook/react/blob/77c4ac2ce88736bbdfe0b29008b5df931c2beb1e/packages/react-reconciler/src/ReactWorkTags.js#L45C14-L45C27) in React).

When an element `type` capitalized is `User-Defined Component` of JSX (which is then represented as a `User-Defined Component` or a pre-defined `component` (e.g.: `Profiler`) in `React`)

See: https://legacy.reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized